basescript/basescript.py

```diff
@@ -31,17 +32,26 @@ def __init__(self):
         self.args = self.parser.parse_args()
 
         self.hostname = socket.gethostname()
-        self.log = self.init_logger(self.args.log_level)
+        self.log = self.init_logger()
 
         args = { n: getattr(self.args, n) for n in vars(self.args) }
+        args['func'] = self.args.func.func_name
         self.log.debug("basescript init", **args)
 
     def start(self):
         '''
         Starts execution of the script
         '''
         # invoke the appropriate sub-command as requested from command-line
-        self.args.func()
+        try:
+            self.args.func()
+        except:
```

Comment:
@algrebe when we catch `SystemExit` for sys.exit(0), we should have same flow as successful completion.

@algrebe For `KeyboardInterrupt` - I see what you are saying. Let us just have it exit with return code 1.